module/zfs/zfs_sysfs.c

```diff
@@ -184,7 +187,7 @@ zfs_kobj_init(zfs_mod_kobj_t *zkobj, int attr_cnt, int child_cnt,
 		    KM_SLEEP);
 		if (zkobj->zko_children == NULL) {
 			if (attr_cnt > 0) {
-				kmem_free(zkobj->zko_attr_list,
+				kmem_free(zkobj->zko_default_attrs,
```

Comment:
Good catch.  Although `kmem_zalloc` is unlikely to fail here, this path is subject to the same leak as was fixed above and should probably have separate checks here as well.

Reply:
Yes indeed!

Reply:
>  Although kmem_zalloc is unlikely to fail here...

Building on @don-brady's point, since we are using `KM_SLEEP`, is it even necessary to handle the case where this allocation fails?

Reply:
Not really since the allocation can't fail as things are currently implemented.  But since @don-brady was nice enough to add the error handling I didn't see a reason to remove it.